perceval/utils/statevector.py

```diff
@@ -78,6 +78,19 @@ def allstate_iterator(input_state: Union[BasicState, StateVector], mask=None) ->
         for output_state in output_array:
             yield BasicState(output_state)
 
+def bs_iterator(m: int, n_max: int):
```

Comment:
Care that bs also means beam splitter in Perceval. Here you're referring to basic state, I guess.

I'd rather name this function `max_photon_state_iterator` or something?

Reply:
Done !